src/utils/mod.rs

```diff
@@ -610,6 +611,73 @@ pub enum PublicIdUnion {
     Ecc(Box<TPMS_ECC_POINT>),
 }
 
+// The digests field type is from bindgen which does not implement Debug on them.
+#[allow(missing_debug_implementations)]
+#[derive(Clone, Default)]
+pub struct DigestList {
+    digests: Vec<Digest>,
+}
+
+impl DigestList {
+    pub fn new() -> Self {
+        DigestList {
+            digests: Vec::new(),
+        }
+    }
+
+    pub fn value(&self) -> &[Digest] {
+        &self.digests
+    }
+
+    pub fn add(&mut self, dig: Digest) -> Result<()> {
+        if self.digests.len() >= 8 {
+            return Err(Error::local_error(WrapperErrorKind::WrongParamSize));
+        }
+        self.digests.push(dig);
+        Ok(())
+    }
+}
+
+impl TryFrom<TPML_DIGEST> for DigestList {
+    type Error = Error;
+    fn try_from(tpml_digest: TPML_DIGEST) -> Result<Self> {
+        let digests_count = tpml_digest.count as usize;
+        if digests_count < 2 {
+            error!("Error: Invalid TPML_DIGEST count(< 2)");
+            return Err(Error::local_error(WrapperErrorKind::InvalidParam));
+        }
+        if digests_count > 8 {
+            error!("Error: Invalid TPML_DIGEST count(> 8)");
+            return Err(Error::local_error(WrapperErrorKind::InvalidParam));
+        }
+        let digests = &tpml_digest.digests[..digests_count];
+        let digests: Result<Vec<Digest>> = digests.iter().map(|x| Digest::try_from(*x)).collect();
+        Ok(DigestList { digests: digests? })
+    }
+}
+
+impl TryFrom<DigestList> for TPML_DIGEST {
+    type Error = Error;
+    fn try_from(digest_list: DigestList) -> Result<Self> {
+        if digest_list.digests.len() < 2 {
+            return Err(Error::local_error(WrapperErrorKind::WrongParamSize));
+        }
+        if digest_list.digests.len() > 8 {
+            return Err(Error::local_error(WrapperErrorKind::WrongParamSize));
+        }
```

Comment:
Given the restrictions placed within the `DigestList` type, are these checks actually needed? Or is it more of a safety net in case the code above changes?

Reply:
That was intended more as a safety net.

Reply:
And for consistency with the other `TryFrom` implementations that also perform checks on the values.